### DISCUSSION TITLE
How to create a function to use in avg_comparisons( transform=)

### DISCUSSION THREAD

**Discussion by daSilva5** [vincentarelbundock/marginaleffects#1306] [Q&A]
I have a model with a transformed response.

I have tried the following:
```back_boxcox <- function(x, roundlam, offset) {
  return ((roundlam * x + 1)^(1 / roundlam) - offset)
}

offset<-1

compare_weight<-avg_comparisons(bbiom,variables = list(treatment = "pairwise",transform=function(x) back_boxcox (x, roundlam = p1$roundlam, offset = 1)))
Warning message:
These variables were not found: transform.  Try specifying the `newdata` argument
  explicitly and make sure the missing variable is included. 
```

I tried to add the newdata, specify the response variable directly, but none of these worked. And could not find examples of how to set up a new back transformation with this package.

Thanks

**Comment by vincentarelbundock** [comment]
It's hard to help without a [fully reproducible MINIMAL working example](https://marginaleffects.com/bonus/help.html#getting-help)

But at first glance, I see that you have mismatched parentheses: `transform` should be on its own, and *not* inside the `variables=list()`.

Also, probably no space after `back_boxcox`

**Comment by vincentarelbundock** [comment]
Also, note that `transform` will apply as the very last operation. You can do something more complicated by defining your own function in the `comparison` argument. See the documentation for examples:

https://marginaleffects.com/r/man/comparisons.html

**Comment by daSilva5** [comment]
Hi,
Removing the space did work! Specifying it in `comparison`  produced the desired result.

```
back_boxcox <- function(x, roundlam, offset) {
  return ((roundlam * x + 1)^(1 / roundlam) - offset)
}
avg_comparisons(bbiom,comparison = function(hi, lo)mean(back_boxcox(hi,roundlam=p1$roundlam,offset=1))-mean(back_boxcox(lo,roundlam=p1$roundlam,offset=1)))
```
Cheers,
